src/pydvl/influence/torch/functional.py

```diff
@@ -201,55 +248,540 @@ def get_hvp_function(
         data_loader: A DataLoader instance that provides batches of data for calculating the Hessian-vector product.
             Each batch from the DataLoader is assumed to return a tuple where the first element
             is the model's input and the second element is the target output.
-        use_hessian_avg: If True, the returned function uses batch-wise Hessian computation via
+        precompute_grad: If True, the full data gradient is precomputed and kept in memory, which can speed up
+            the hessian vector product computation. Set this to False, if you can't afford to keep an additional
+            parameter-sized vector in memory.
+        use_average: If True, the returned function uses batch-wise computation via
             [batch_loss_function][pydvl.influence.torch.functional.batch_loss_function] and averages the results.
             If False, the function uses backpropagation on the full
             [empirical_loss_function][pydvl.influence.torch.functional.empirical_loss_function],
             which is more accurate than averaging the batch hessians, but probably has a way higher memory usage.
         reverse_only: Whether to use only reverse-mode autodiff (True, default) or
```

Comment:
Types and default values are redundant in docstrings (there are many instances, just pointing this one out)

```suggestion
        reverse_only: Whether to use only reverse-mode autodiff or
```